raiden/network/proxies/token_network_registry.py

```diff
@@ -113,107 +115,332 @@ def add_token(
         if token_address == NULL_ADDRESS_BYTES:
             raise InvalidTokenAddress("The call to register a token at 0x00..00 will fail.")
 
-        # check preconditions
+        if token_network_deposit_limit <= 0:
+            raise InvalidTokenNetworkDepositLimit(
```

Comment:
I heard that the proxies should not raise specific errors.  This should be `BrokenPreconditionError`.  https://github.com/raiden-network/raiden/issues/4878#issuecomment-531728260

Reply:
@pirapira No... it seems like the definition is still vague. This should not be a precondition error.

Reply:
`token_network_deposit_limit > 0` is a precondition.  Here we're in a proxy and checking a precondition before estimating the gas.  So the exception should be `BrokenPreconditionError`.

Reply:
This point is kept track in https://github.com/raiden-network/raiden/pull/4957